catroid/src/main/java/org/catrobat/catroid/transfers/project/ProjectUploadService.kt

```diff
@@ -0,0 +1,206 @@
+            archiveDirectory = File(cacheDir, UPLOAD_FILE_NAME),
+            zipArchiver = ZipArchiver(),
+            screenshotLoader = ProjectAndSceneScreenshotLoader(
+                applicationContext.resources.getDimensionPixelSize(R.dimen.project_thumbnail_width),
+                applicationContext.resources.getDimensionPixelSize(R.dimen.project_thumbnail_height)
+            ),
+            sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this),
+            serverCalls = ServerCalls.getInstance()
+        ).start(
+            successCallback = { projectId ->
+                Log.v(TAG, "Upload successful")
+                stopForeground(true)
+                val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as? NotificationManager
+                notificationManager?.notify(UPLOAD_NOTIFICATION_ID, createUploadFinishedNotification(projectName))
+
+                ToastUtil.showSuccess(this, R.string.notification_upload_finished)
+                val result = Bundle().apply { putInt(Constants.EXTRA_PROJECT_ID, projectId) }
+                resultReceiver.send(UPLOAD_RESULT_RECEIVER_RESULT_CODE, result)
+            },
+            errorCallback = { errorCode, errorMessage ->
+                Log.e(TAG, errorMessage)
+                stopForeground(true)
+                ToastUtil.showError(this, resources.getString(R.string.error_project_upload) + " " + errorMessage)
+                StatusBarNotificationManager.getInstance()
+                    .createUploadRejectedNotification(applicationContext, errorCode, errorMessage, reUploadBundle)
+            }
+        )
+    }
+
+    override fun onDestroy() {
+        Utils.invalidateLoginTokenIfUserRestricted(applicationContext)
+        super.onDestroy()
+    }
+
+    private fun createUploadNotification(programName: String): Notification {
+        StatusBarNotificationManager.getInstance().createNotificationChannel(applicationContext)
+
+        var uploadIntent = Intent(applicationContext, MainMenuActivity::class.java)
+        uploadIntent.action = Intent.ACTION_MAIN
+        uploadIntent = uploadIntent.setFlags(Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED)
+        val pendingIntent = PendingIntent.getActivity(
+            applicationContext,
+            StatusBarNotificationManager.UPLOAD_PENDING_INTENT_REQUEST_CODE,
+            uploadIntent, PendingIntent.FLAG_CANCEL_CURRENT
+        )
+
+        return NotificationCompat.Builder(applicationContext, StatusBarNotificationManager.CHANNEL_ID)
+            .setContentIntent(pendingIntent)
+            .setSmallIcon(R.drawable.ic_stat)
+            .setContentTitle(
+                applicationContext.getString(R.string.notification_upload_title_pending) + " " + programName
+            )
+            .setContentText(applicationContext.getString(R.string.notification_upload_pending))
+            .setOngoing(true)
+            .setProgress(MAX_PERCENT, 0, true)
```

Comment:
could you set the MAX_PERCENT also in the StatusBarNotificationManager.showOrUpdateNotification, because here also 100 is used and a beautiful constant could be useful for all uses of setProgress()

Reply:
marking this resolved since it refers to other parts of the code